tsfc/coffee.py

```diff
@@ -320,11 +334,20 @@ def _expression_scalar(expr, parameters):
     if isnan(expr.value):
         return coffee.Symbol("NAN")
     else:
-        v = expr.value
-        r = round(v, 1)
-        if r and abs(v - r) < parameters.epsilon:
-            v = r  # round to nonzero
-        return coffee.Symbol(("%%.%dg" % parameters.precision) % v)
+        vr = expr.value.real
+        rr = round(1)
+        if rr and abs(vr - rr) < parameters.epsilon:
+            vr = rr  # round to nonzero
+
+        vi = expr.value.imag  # also checks if v is purely real
+        if vi == 0.0:
+            return coffee.Symbol(("%%.%dg" % parameters.precision) % vr)
+        ri = round(1)
+
+        if ri and abs(vi - ri) < parameters.epsilon:
+            vi = ri
+        return coffee.Symbol(("%%.%dg" % parameters.precision) % vr + " + " +
+                             ("%%.%dg" % parameters.precision) % vi + " * I")
```

Comment:
Looks slightly dangerous... COFFEE probably does not surround symbols with parenthesis, which can lead to a precedence issue.